Segunda entrega - Trabajo final

Integrantes:
*   Luis Armando Ibatá Ramirez
*   Juan Pablo Román Patiño
*   Gabriela Cuervo Rojas


# CARGA DE DATOS

In [ ]:
import pandas as pd 
import numpy as np
import plotly.express as px
from plotly.subplots import make_subplots
import io

Para sacar el nombre de la semana acc['FECHA2'].dt.strftime('%a')

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
##Base accidentalidad en el municipio de Palmira
df1 = pd.read_csv('/content/drive/MyDrive/analitica/Accidentes_de_transito_Palmira_2020.csv',encoding='latin-1', sep = ';') 
df1.head()

GRAVEDAD           FECHA   AÑO                     HORA JORNADA DIA_SEMANA  \
0    DAÑOS  1/01/2020 0:00  2020  1899-12-31T19:10:00.000   NOCHE  MIERCOLES   
1    DAÑOS  1/01/2020 0:00  2020  1899-12-31T19:10:00.000   NOCHE  MIERCOLES   
2    DAÑOS  1/02/2020 0:00  2020  1899-12-31T12:10:00.000   TARDE     JUEVES   
3    DAÑOS  1/02/2020 0:00  2020  1899-12-31T12:10:00.000   TARDE     JUEVES   
4  HERIDOS  1/01/2020 0:00  2020  1899-12-31T12:20:00.000   TARDE  MIERCOLES   

  BARRIOS-COREGIMIENTO- VIA                 DIRECCION    ZONA AUTORIDAD  ...  \
0                  OLIMPICO     CALLE 23 CARRERA 35 A  URBANA   TECNICO  ...   
1                  OLIMPICO     CALLE 23 CARRERA 35 A  URBANA   TECNICO  ...   
2                INDUSTRIAL       CALLE 42 CARRERA 35  URBANA   TECNICO  ...   
3                INDUSTRIAL       CALLE 42 CARRERA 35  URBANA   TECNICO  ...   
4                  ZAMORANO  CARRERA 28 CALLE 664A-19  URBANA   TECNICO  ...   

   CLASE DE SINIESTRO  LESIONADO HOMICIDIOS    CLINICA      SITIO  \
0              CHOQUE  NO APLICA  NO APLICA  NO APLICA  NO APLICA   
1              CHOQUE  NO APLICA  NO APLICA  NO APLICA  NO APLICA   
2              CHOQUE  NO APLICA  NO APLICA  NO APLICA  NO APLICA   
3              CHOQUE  NO APLICA  NO APLICA  NO APLICA  NO APLICA   
4              CHOQUE          1  NO APLICA      PALMA  NO APLICA   

  CLASE VEHICULO      MARCA     MATRICULA TIPO DE SERVICIO    EMPRESA  
0      AUTOMOVIL  CHEVROLET          CALI       PARTICULAR  NO APLICA  
1      AUTOMOVIL        KIA          CALI       PARTICULAR  NO APLICA  
2      CAMIONETA  CHEVROLET  CUNDINAMARCA       PARTICULAR  NO APLICA  
3      AUTOMOVIL     DAEWOO          CALI       PARTICULAR  NO APLICA  
4           MOTO      HONDA       PRADERA       PARTICULAR  NO APLICA  

[5 rows x 24 columns]

In [ ]:
##Base accidentalidad en el municipio de Envigado
df2 = pd.read_csv('/content/drive/MyDrive/analitica/Envigado.csv',encoding='latin-1', sep = ';')
df2.head()

RADICADO      FECHA             HORA DÍA DE LA SEMANA CLASE DE VEHICULO  \
0   2200001  1/01/2020  sin información        Miércoles         Camioneta   
1   2200002  1/01/2020  sin información        Miércoles         Automovil   
2   2200002  1/01/2020  sin información        Miércoles       Motocicleta   
3   2200003  1/01/2020  sin información        Miércoles         Camioneta   
4   2200003  1/01/2020  sin información        Miércoles       Motocicleta   

  TIPO DE SERVICIO TIPO DE VICTIMA          SEXO  ESTADO DE BEODEZ  \
0       PARTICULAR       Conductor  Sin registro                 0   
1       PARTICULAR       Conductor  Sin registro                 0   
2       PARTICULAR    Motociclista  Sin registro                 0   
3       PARTICULAR       Conductor  Sin registro                 0   
4       PARTICULAR    Sin Registro      Femenino                 0   

   RESULTADO DE BEODEZ    GRAVEDAD CLASE DE ACCIDENTE  \
0                    0  Solo latas        Volcamiento   
1                    0  Solo latas             Choque   
2                    0  Solo latas             Choque   
3                    0  Solo latas             Choque   
4                    0     Heridos             Choque   

                                      CAUSA                     DIRECCIÓN  \
0  Falta de precaucion niebla. lluvia. humo              KILOMETRO 13+750   
1                       Desobedecer señales  CALLE 40 SUR CARRERA 24 F 91   
2                       Desobedecer señales  CALLE 40 SUR CARRERA 24 F 91   
3                    Impericia en el manejo   CARRERA 42 CALLE 41B SUR 11   
4                    Impericia en el manejo   CARRERA 42 CALLE 41B SUR 11   

              BARRIO    AREA                Coordenadas  
0  Vereda las palmas   Rural                        NaN  
1          El Salado  Urbana    (61507253, 15072529968)  
2          El Salado  Urbana    (61507253, 15072529968)  
3          El Dorado  Urbana  (6165552426, 16555242646)  
4          El Dorado  Urbana  (6165552426, 16555242646)

In [ ]:
##Base accidentalidad en la ciudad de Medellin
df3 = pd.read_csv('/content/drive/MyDrive/analitica/DATOS 2020.csv',encoding='latin-1', sep = ';') 
df3.head(5)

AÑO    CBML CLASE_ACCIDENTE      DIRECCION       DIRECCION ENCASILLADA  \
0  2020     5.0          Choque  CR 64 C CL 78  CR  064 C   078  000 00000   
1  2020  1307.0       Atropello  CL 49 CR 99 E  CL  049   099 E  000 00000   
2  2020  6086.0          Choque   CL 73 CR 152    CL  073   152  000 00000   
3  2020   710.0            Otro    CR 80 CL 80    CR  080   080  000 00000   
4  2020   914.0          Choque    CR 35 CL 29    CR  035   029  000 00000   

         DISEÑO  EXPEDIENTE  FECHA_ACCIDENTE GRAVEDAD_ACCIDENTE  MES  \
0   Tramo de via  A001080297  8/01/2020 21:10        Solo daños    1   
1   Tramo de via  A001080125  8/01/2020 21:45        Con heridos    1   
2   Tramo de via  A001080291  8/01/2020 23:00        Solo daños    1   
3       Glorieta  A001080128   9/01/2020 4:50        Con heridos    1   
4  Lote o Predio  A001080130   9/01/2020 5:00        Con heridos    1   

   NRO_RADICADO NUMCOMUNA                  BARRIO  \
0       1703041         5  Terminal de Transporte   
1       1702988        13   Juan XXIII La Quiebra   
2       1703048        60   Suburbano La Cuchilla   
3       1702992         7          López de Mesa   
4       1702984         9         Asomadera No. 1   

                            COMUNA                         LOCATION  \
0                         Castilla  [-75.5729956763, 6.27615598793]   
1                       San Javier   [-75.613806632, 6.26637688097]   
2  Corregimiento de San Cristóbal  [-75.6515379855, 6.29139512193]   
3                          Robledo  [-75.5843078051, 6.28437222664]   
4                     Buenos Aires  [-75.5632244526, 6.22711885945]   

           X           Y  
0  834495.75  1186012.91  
1  829975.02  1184944.15  
2  825806.52  1187724.26  
3  833246.18  1186925.42  
4  835562.04  1180585.15

# TRATAMIENTO DE DATOS

In [ ]:
df1.columns

Index(['GRAVEDAD', 'FECHA', 'AÑO', 'HORA', 'JORNADA', 'DIA_SEMANA',
       'BARRIOS-COREGIMIENTO- VIA', 'DIRECCION', 'ZONA', 'AUTORIDAD', 'LAT',
       'LONG', 'HIPOTESIS', 'CONDICION DE LA VICTIMA', 'CLASE DE SINIESTRO',
       'LESIONADO', 'HOMICIDIOS', 'CLINICA', 'SITIO', 'CLASE VEHICULO',
       'MARCA', 'MATRICULA', 'TIPO DE SERVICIO', 'EMPRESA'],
      dtype='object')

In [ ]:
df2.columns

Index(['RADICADO', 'FECHA', 'HORA', 'DÍA DE LA SEMANA', 'CLASE DE VEHICULO',
       'TIPO DE SERVICIO', 'TIPO DE VICTIMA', 'SEXO', 'ESTADO DE BEODEZ',
       'RESULTADO DE BEODEZ', 'GRAVEDAD', 'CLASE DE ACCIDENTE', 'CAUSA',
       'DIRECCIÓN', 'BARRIO', 'AREA', 'Coordenadas'],
      dtype='object')

In [ ]:
df3.columns

Index(['AÑO', 'CBML', 'CLASE_ACCIDENTE', 'DIRECCION', 'DIRECCION ENCASILLADA',
       'DISEÑO', 'EXPEDIENTE', 'FECHA_ACCIDENTE', 'GRAVEDAD_ACCIDENTE', 'MES',
       'NRO_RADICADO', 'NUMCOMUNA', 'BARRIO', 'COMUNA', 'LOCATION', 'X', 'Y'],
      dtype='object')

**Convertir las fechas de las tres bases de datos df1, df2 y df3 en formato datetime**

In [ ]:
df1['FECHA']=pd.to_datetime(df1['FECHA'])

In [ ]:
##Se usa la funcion dt.date para separar la hora de la fecha de la base de datos df3
df3['FECHA_ACCIDENTE']=pd.to_datetime(df3['FECHA_ACCIDENTE']).dt.date

In [ ]:
df3['FECHA_ACCIDENTE']=pd.to_datetime(df3['FECHA_ACCIDENTE'])

In [ ]:
df2['FECHA']=pd.to_datetime(df2['FECHA'])

In [ ]:
##Se renombran las columnas que tienen en comun las tres bases, para que todas queden con el mismo nombre
df3 = df3.rename(columns ={'GRAVEDAD_ACCIDENTE':'GRAVEDAD'})
df2 = df2.rename(columns ={'CLASE DE ACCIDENTE':'CLASE_ACCIDENTE'})
df1 = df1.rename(columns ={'CLASE DE SINIESTRO':'CLASE_ACCIDENTE'})
df3 = df3.rename(columns ={'FECHA_ACCIDENTE':'FECHA'})

**Para que todas las categorias de la columna GRAVEDAD de cada base queden iguales**

In [ ]:
df1['GRAVEDAD'].unique()

array(['DAÑOS', 'HERIDOS', 'MUERTO'], dtype=object)

In [ ]:
df2['GRAVEDAD'].unique()

array(['Solo latas', 'Heridos', 'Muertos'], dtype=object)

In [ ]:
df3['GRAVEDAD'].unique()

array(['Solo daños', 'Con heridos', 'Con muertos'], dtype=object)

In [ ]:
df2['GRAVEDAD']=df2['GRAVEDAD'].replace({'Solo latas':'DAÑOS','Heridos':'HERIDOS','Muertos':'MUERTO'})
df3['GRAVEDAD']=df3['GRAVEDAD'].replace({'Solo daños':'DAÑOS','Con heridos':'HERIDOS','Con muertos':'MUERTO'})

**Para que todas las categorias de la columna CLASE_ACCIDENTE de cada base queden iguales**

In [ ]:
df1['CLASE_ACCIDENTE'].unique()

array(['CHOQUE', 'VOLCAMIENTO', 'CHOQUE OBJETO FIJO', 'ATROPELLO',
       'CHOQUE SEMOVIENTE', 'OTRO', 'CHOQUE MURO'], dtype=object)

In [ ]:
df2['CLASE_ACCIDENTE'].unique()

array(['Volcamiento', 'Choque', 'Atropello', 'Caida ocupante', 'Otro',
       'Incendio'], dtype=object)

In [ ]:
df3['CLASE_ACCIDENTE'].unique()

array(['Choque', 'Atropello', 'Otro', 'Volcamiento', 'Caida Ocupante',
       'Incendio'], dtype=object)

**Se colocan todas las categorias de CLASE_ACCIDENTE en minúsculas**

In [ ]:
df1['CLASE_ACCIDENTE']=df1['CLASE_ACCIDENTE'].replace({'CHOQUE OBJETO FIJO':'CHOQUE','CHOQUE SEMOVIENTE':'CHOQUE','CHOQUE MURO':'CHOQUE'})
df1['CLASE_ACCIDENTE']=df1['CLASE_ACCIDENTE'].map(str.lower)
df2['CLASE_ACCIDENTE']=df2['CLASE_ACCIDENTE'].map(str.lower)
df3['CLASE_ACCIDENTE']=df3['CLASE_ACCIDENTE'].map(str.lower)

**Se procede a analizar la cantidad de nulos que tiene cada base de datos**

In [ ]:
df1.isnull().sum()

GRAVEDAD                     0
FECHA                        0
AÑO                          0
HORA                         0
JORNADA                      0
DIA_SEMANA                   0
BARRIOS-COREGIMIENTO- VIA    0
DIRECCION                    0
ZONA                         0
AUTORIDAD                    0
LAT                          0
LONG                         0
HIPOTESIS                    0
CONDICION DE LA VICTIMA      0
CLASE_ACCIDENTE              0
LESIONADO                    0
HOMICIDIOS                   0
CLINICA                      0
SITIO                        0
CLASE VEHICULO               0
MARCA                        0
MATRICULA                    0
TIPO DE SERVICIO             0
EMPRESA                      0
dtype: int64

In [ ]:
df2.isnull().sum()

RADICADO                 0
FECHA                    0
HORA                     0
DÍA DE LA SEMANA         0
CLASE DE VEHICULO        0
TIPO DE SERVICIO         0
TIPO DE VICTIMA          0
SEXO                     0
ESTADO DE BEODEZ         0
RESULTADO DE BEODEZ      0
GRAVEDAD                 0
CLASE_ACCIDENTE          0
CAUSA                    0
DIRECCIÓN                0
BARRIO                   0
AREA                     0
Coordenadas            216
dtype: int64

In [ ]:
df3.isnull().sum()

AÑO                        0
CBML                     242
CLASE_ACCIDENTE            0
DIRECCION                  0
DIRECCION ENCASILLADA     15
DISEÑO                    0
EXPEDIENTE                 0
FECHA                      0
GRAVEDAD                   0
MES                        0
NRO_RADICADO               0
NUMCOMUNA                  0
BARRIO                   243
COMUNA                   242
LOCATION                   0
X                          0
Y                          0
dtype: int64

**En esta base df3, si se tiene un número de nulos considerable pero no se pueden reemplazar, ya que lo que falta del CBML son los mismos datos que faltan de la comuna, por lo tanto no se pueden agrupar o rellenar con información, entonces como no se tiene certeza donde fue el accidente se procede a llenarlos con Sin Inf.**

In [ ]:
df3 = df3.fillna('Sin Inf') #Se cambian los nulos por la palabra sin inf

**Para unir las 3 bases de datos, por las columnas que comparten que son Fecha, Gravedad y Clase de Accidente, se procede a crear una serie de tiempo para cada una de las ciudades por la fecha para asi estar seguros que se analizaran todas las fechas de cada base (df1,df2,df3)**

In [ ]:
df4= pd.DataFrame(pd.date_range(start=df1['FECHA'].min(), end=df1['FECHA'].max())).rename(columns ={0:'FECHA'}) ##Serie de Tiempo de la ciudad de Palmira
df4

FECHA
0   2020-01-01
1   2020-01-02
2   2020-01-03
3   2020-01-04
4   2020-01-05
..         ...
361 2020-12-27
362 2020-12-28
363 2020-12-29
364 2020-12-30
365 2020-12-31

[366 rows x 1 columns]

In [ ]:
df5= pd.DataFrame(pd.date_range(start=df2['FECHA'].min(), end=df2['FECHA'].max())).rename(columns ={0:'FECHA'})##Serie de tiempo Envigado
df5

FECHA
0   2020-01-01
1   2020-01-02
2   2020-01-03
3   2020-01-04
4   2020-01-05
..         ...
342 2020-12-08
343 2020-12-09
344 2020-12-10
345 2020-12-11
346 2020-12-12

[347 rows x 1 columns]

In [ ]:
df6= pd.DataFrame(pd.date_range(start=df3['FECHA'].min(), end=df3['FECHA'].max())).rename(columns ={0:'FECHA'})##Serie de tiempo Medellin
df6

FECHA
0   2020-01-01
1   2020-01-02
2   2020-01-03
3   2020-01-04
4   2020-01-05
..         ...
338 2020-12-04
339 2020-12-05
340 2020-12-06
341 2020-12-07
342 2020-12-08

[343 rows x 1 columns]

**Se procede a unir la base de datos original con la serie de tiempo correspondiente, para esto se utiliza la funcion merge**

In [ ]:
df7= pd.merge(df4, df1, on ='FECHA', how ='left')##Palmira

In [ ]:
df8= pd.merge(df5, df2, on ='FECHA', how ='left')##Envigado
df9= pd.merge(df6, df3, on ='FECHA', how ='left')##Medellin

**Luego se procede a agrupar la cantidad de accidentes por la fecha, gravedad y la clase de accidente y se hace un count para contar la cantidad de accidentes que hubo**

In [ ]:
palmira=df7.groupby(['FECHA','GRAVEDAD','CLASE_ACCIDENTE'])[['JORNADA']].count().reset_index()
envigado=df8.groupby(['FECHA','GRAVEDAD','CLASE_ACCIDENTE'])[['RADICADO']].count().reset_index()
medellin=df9.groupby(['FECHA','GRAVEDAD','CLASE_ACCIDENTE'])[['NUMCOMUNA']].count().reset_index()

**Se agrupan las tres bases de datos con el merge para observar la cantidad de accidentes que hubieron en la ciudad de Palmira, Envigado y Medellin dependiendo de la fecha**

In [ ]:
bodega=pd.merge(palmira,envigado, on=['FECHA','GRAVEDAD','CLASE_ACCIDENTE'], how='outer').merge(medellin, on=['FECHA','GRAVEDAD','CLASE_ACCIDENTE'], how='outer').rename(columns ={'JORNADA':'PALMIRA','RADICADO':'ENVIGADO','NUMCOMUNA':'MEDELLIN'})

In [ ]:
bodega= bodega.fillna(0.0)  #se cambian los nulos de la bodega de datos anterior por ceros

In [ ]:
bodega.head()

FECHA GRAVEDAD CLASE_ACCIDENTE  PALMIRA  ENVIGADO  MEDELLIN
0 2020-01-01    DAÑOS          choque      2.0       3.0      18.0
1 2020-01-01  HERIDOS          choque      3.0       2.0      26.0
2 2020-01-02    DAÑOS          choque      4.0       5.0      62.0
3 2020-01-03    DAÑOS          choque      8.0       2.0      27.0
4 2020-01-03  HERIDOS          choque      7.0       2.0      19.0

# PREGUNTA 1: ¿CÚAL ES LA CANTIDAD DE ACCIDENTES SEGÚN EL TIPO DE GRAVEDAD EN LAS 3 CIUDADES?

In [ ]:
# se crea un datafrime nuevo igual a bodega pero  sin la fecha
HOLA = bodega.groupby(['GRAVEDAD'])[['PALMIRA','ENVIGADO','MEDELLIN']].sum().reset_index()
HOLA

GRAVEDAD  PALMIRA  ENVIGADO  MEDELLIN
0    DAÑOS    960.0    1349.0    7651.0
1  HERIDOS   1380.0     405.0   11959.0
2   MUERTO     40.0       7.0     107.0

In [ ]:
# definir gráfica
fig = px.bar(HOLA, x='GRAVEDAD', y=['PALMIRA','ENVIGADO','MEDELLIN'], barmode= 'group', title ='<b>Accidentes segun el tipo de ciudad<b>')
            
px.bar()
# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Tipo de Gravedad',
    yaxis_title = 'Cantidad de Accidentes',
    template = 'simple_white',
    title_x = 0.5,
    legend_title = '<b>Ciudades<b>')


fig.show()

En las tres categorías domina Medellín con más de 7000 datos en DAÑOS Y HERIDOS, esto se puede dar posiblemente por que la extensión en área de Medellín es superior a la de Palmira y Envigado, ya que estas dos en ninguna categoría supera los 2000 datos. 

En DAÑOS, Envigado supera a Palmira, pero no de manera significativa, ya que la diferencia entre ambas no llega a los 500 datos, pero en HERIDOS es todo lo contrario, Palmira sobrepasa a Envigado notablemente, casi por los 1000 datos, es posible que esto suceda porque los accidentes en Palmira son más fuertes o las personas son más imprudentes que Envigado.

Por el lado de los muertos, si se ve que en las tres casi no suceden siniestros fatales con relación a las demás categorías, ya que el dato máximo es de 107 y el mínimo es de 7, lo que significa que tiene una proporciona muy mínima lo cual en cuestión de indicadores se puede decir que es bueno.


# PREGUNTA 2: ¿CÚAL ES LA CLASE DE ACCIDENTE QUE MAS REGISTROS TIENEN SEGÚN LA EVOLUCION DIARIA EN LAS 3 CIUDADES?

In [ ]:
#Se crea un dataframe tomando la fecha pero tomando solo el mes y año, clase de accidente y se cuenta con la columna gravedad
bdb=bodega.groupby([bodega['FECHA'].dt.strftime('%m-%Y'),'CLASE_ACCIDENTE'])[['GRAVEDAD']].count().reset_index()  
bdb

FECHA CLASE_ACCIDENTE  GRAVEDAD
0   01-2020       atropello        33
1   01-2020  caida ocupante        29
2   01-2020          choque        68
3   01-2020            otro        32
4   01-2020     volcamiento        35
..      ...             ...       ...
58  12-2020       atropello        14
59  12-2020  caida ocupante         8
60  12-2020          choque        63
61  12-2020            otro        11
62  12-2020     volcamiento        10

[63 rows x 3 columns]

In [ ]:
#Se realiza el grafico del dataframe anterior   
fig = px.area(bdb, x = 'FECHA', y ='GRAVEDAD', color = 'CLASE_ACCIDENTE',
             title= '<b>Cantidad de clases de accidentes que ocurrieron en el 2020<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

fig.update_layout(
    xaxis_title = 'Mes-Año',
    yaxis_title = 'Accidentes',
    template = 'simple_white',
    title_x = 0.5)
fig.show()

Se observa que la categoría de choque es la que más datos registra en comparación a las demás y esto se mantiene durante el trascurrir de los meses, luego se puede decir que le siguen volcamiento, incendio y otro, estos tres con una cantidad de datos aproximadamente iguales o con muy poca diferencia entre ellos.

Por último, esta caída de ocupante que es la que menos datos tiene dentro de todas las categorías mencionadas.

Para todos los meses se mantiene una relación directa entre todas las categorías, ya que cuando una se mantiene constante, las demás se comportan constantes, y cuando baja una las demás también lo hacen


# PREGUNTA 3: ¿CÚAL ES EL PORCENTAJE DE ACCIDENTES QUE HAY EN LAS TRES CIUDADES DEPENDIENDO DE LA CLASE DEL ACCIDENTE?

In [ ]:
#Se crea un dataframe con clase de accidente y gravedad
total=bodega.groupby(['CLASE_ACCIDENTE'])[['GRAVEDAD']].count().reset_index()
total

CLASE_ACCIDENTE  GRAVEDAD
0       atropello       296
1  caida ocupante       253
2          choque       765
3        incendio         3
4            otro       276
5     volcamiento       279

In [ ]:
#Se realiza el grafico de dataframe total
fig = px.pie(total , values = 'GRAVEDAD', names = 'CLASE_ACCIDENTE', title = '<b>% Accidentes segun el tipo de accidente<b>')
# agregar detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    legend_title = 'Clase de Accidente',
    title_x = 0.5)

fig.show()

Según la clase de accidente en las tres ciudades, decimos que choque es la que más sobresale, casi con la mitad de los datos, y en comparación con las demás categorías esta las supera por mucho, con más de un 25% aproximadamente.

En las categorías restantes exceptuando a incendio no se logra ver una diferencia significativa entre alguna de ellas, ya que otro, atropello, volcamiento y caída de ocupante oscilan entre el 13% y el 16%, estos se encuentran relativamente cercanos.

Incendio es el que no tiene mucha relevancia en estos datos ya que tiene solo un 0.16%.


# PREGUNTA 4: ¿CÚAL ES LA PROPORCIÓN DE ACCIDENTES EN LA CIUDAD DE ENVIGADO TENIENDO EN CUENTA EL SEXO?

In [ ]:
sexo=df8.groupby(['SEXO'])[['RADICADO']].count().reset_index().rename(columns={'RADICADO':'ACCIDENTES'})
sexo

SEXO  ACCIDENTES
0      Femenino         134
1     Masculino         277
2  Sin registro        1350

In [ ]:
#Se realiza el grafico de dataframe total
fig = px.pie(sexo, values = 'ACCIDENTES', names = 'SEXO', title = '<b>% Accidentes segun el tipo de accidente<b>')
# agregar detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    legend_title = 'Clase de Accidente',
    title_x = 0.5)

fig.show()

Teniendo en cuenta el sexo en los accidentes, podemos ver que la categoría sin registro es la que predomina con un 76.7%, casi con 2/3 de los datos, lo cual es muy significativo porque nos dice que a la hora de tomar los datos de un accidente la edad no es relevante y puede que la pasen por alto o por el contrario no se sepa con certeza quien iba manejando a la hora del siniestro.

Ya en relación entre mujeres y hombres, podemos decir que los hombres son los que más se accidentan, superando casi por el doble de porcentaje a las mujeres.


# PREGUNTA 5: ¿CÚAL ES LA CANTIDAD DE ACCIDENTES SEGÚN EL TIPO DE ZONA (COMUNA)? TOMANDO SOLO PALMIRA Y ENVIGADO

In [ ]:
#se crea dataframe de palmira con fecha, zona y se cuenta con jornada
zona_palmira=df7.groupby(['FECHA','ZONA'])[['JORNADA']].count().reset_index().rename(columns ={'JORNADA':'PALMIRA'})
zona_palmira.head()

FECHA    ZONA  PALMIRA
0 2020-01-01  URBANA        5
1 2020-01-02   RURAL        2
2 2020-01-02  URBANA        2
3 2020-01-03   RURAL        2
4 2020-01-03  URBANA       13

In [ ]:
df8=df8.rename(columns ={'AREA':'ZONA'}) #se renombra el nombre de AREA

In [ ]:
df8['ZONA']=df8['ZONA'].str.upper() #se convierten todos los datos de la columna ZONA en MAYUSCULA

In [ ]:
df8['ZONA'].unique()

array(['RURAL', 'URBANA', nan], dtype=object)

In [ ]:
df8['ZONA'].isnull().sum()##Los nulos que aparecen es por la serie de tiempo y en esos dias no hubieron accidentes

207

In [ ]:
#se crea dataframe de envigado con fecha, zona y se cuenta con jornada
zona_envigado=df8.groupby(['FECHA','ZONA'])[['RADICADO']].count().reset_index().rename(columns ={'RADICADO':'ENVIGADO'})
zona_envigado.head()

FECHA    ZONA  ENVIGADO
0 2020-01-01   RURAL         1
1 2020-01-01  URBANA         8
2 2020-01-02  URBANA         6
3 2020-01-03  URBANA         7
4 2020-01-04   RURAL         5

In [ ]:
#se concatenan ambos dateframe por fecha y zona
zonas=pd.merge(zona_palmira,zona_envigado, on=['FECHA','ZONA'], how='outer')
zonas.head()

FECHA    ZONA  PALMIRA  ENVIGADO
0 2020-01-01  URBANA      5.0       8.0
1 2020-01-02   RURAL      2.0       NaN
2 2020-01-02  URBANA      2.0       6.0
3 2020-01-03   RURAL      2.0       NaN
4 2020-01-03  URBANA     13.0       7.0

In [ ]:
zonas = zonas.fillna(0.0)  #Se realiza el cambio de los nulos (NaN) por ceros (0.0)

In [ ]:
zonas.head()

FECHA    ZONA  PALMIRA  ENVIGADO
0 2020-01-01  URBANA      5.0       8.0
1 2020-01-02   RURAL      2.0       0.0
2 2020-01-02  URBANA      2.0       6.0
3 2020-01-03   RURAL      2.0       0.0
4 2020-01-03  URBANA     13.0       7.0

In [ ]:
zonas1 = zonas.groupby(['ZONA'])[['PALMIRA','ENVIGADO']].sum().reset_index()
# definir gráfica
fig = px.bar(zonas1, x='ZONA', y=['PALMIRA','ENVIGADO'], barmode= 'group', title ='<b>Accidentes segun el tipo de zona<b>',
            color_discrete_sequence=["#ffdd88", "#88dd44"])

px.bar()

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Tipo de zona',
    yaxis_title = 'Cantidad de Accidentes',
    template = 'simple_white',
    title_x = 0.5,
    legend_title = '<b>Ciudades<b>')


fig.show()

Según el tipo de zona (rural o urbana), se puede observar que la zona rural no tiene muchos accidentes con relación a la urbana, además de que en ambas ciudades con respecto a lo rural los accidentes son prácticamente los mismos.

Tomando ya lo urbano, decimos que Palmira supera a Envigado por más de 600 accidentes, lo cual nos dice que Palmira es un municipio donde hay una posibilidad más alta de sufrir un accidente vial. También podemos decir que una posible causa de esto es la cultura o condición de las vías que se maneja en ambas partes, esto también influye de manera significa al momento de conducir.


# PREGUNTA 6: ¿CÚAL ES LA CANTIDAD DE ACCIDENTES SEGÚN LA CLASE DE VEHICULO EN LAS CIUDADES PALMIRA Y ENVIGADO?

In [ ]:
#Se crea dataframe de clase de vehiculo y se cuentan con la columna radicado, ademas se organiza en forma descendente
clase_vehiculoe=df8.groupby(['CLASE DE VEHICULO'])[['RADICADO']].count().reset_index().sort_values('RADICADO', ascending = False)

In [ ]:
clase_vehiculoe['CLASE DE VEHICULO']=clase_vehiculoe['CLASE DE VEHICULO'].str.upper() #se vuelve toda la columna de clase de vehiculo en MAYUSCULA
clase_vehiculoe['CLASE DE VEHICULO'].unique()

array(['AUTOMOVIL', 'MOTOCICLETA', 'CAMIONETA', 'CAMPERO', 'CAMION',
       'BUS', 'BICICLETA', 'MICROBUS', 'BUSETA', 'TRACTOCAMION',
       'MOTOCARRO', 'VOLQUETA', 'TRICICLO'], dtype=object)

In [ ]:
#Se renombras algunas categorias de la columna clase de vehiculo de la base de datos de palmira para que queden igual a la columna de clase de vehiculo de envigado
df7['CLASE VEHICULO']=df7['CLASE VEHICULO'].replace({'MOTO':'MOTOCICLETA','TRACTO CAMION':'TRACTOCAMION','CICLISTA':'BICICLETA','MAQUINARIA':'NO APLICA'})

In [ ]:
#Se crea dataframe de clase de vehiculo y se cuentan con la columna jornada, ademas se organiza en forma descendente
clase_vehiculop=df7.groupby(['CLASE VEHICULO'])[['JORNADA']].count().reset_index().sort_values('JORNADA', ascending = False)

In [ ]:
clase_vehiculop=clase_vehiculop.rename(columns ={'CLASE VEHICULO':'CLASE DE VEHICULO'}) #se renombra la columna
clase_vehiculop

CLASE DE VEHICULO  JORNADA
9        MOTOCICLETA      924
0          AUTOMOVIL      648
10         NO APLICA      277
5          CAMIONETA      251
4             CAMION       68
6            CAMPERO       61
1          BICICLETA       50
2                BUS       37
11      TRACTOCAMION       19
3             BUSETA       17
7           MICROBUS       17
13          VOLQUETA        7
8          MOTOCARRO        3
12          TRICICLO        1

In [ ]:
#se concatenan por clase de vehiculo
clasevehiculos=pd.merge(clase_vehiculop,clase_vehiculoe, on=['CLASE DE VEHICULO'], how='outer').rename(columns ={'JORNADA':'PALMIRA','RADICADO':'ENVIGADO'})
clasevehiculos

CLASE DE VEHICULO  PALMIRA  ENVIGADO
0        MOTOCICLETA      924     536.0
1          AUTOMOVIL      648     668.0
2          NO APLICA      277       NaN
3          CAMIONETA      251     257.0
4             CAMION       68      76.0
5            CAMPERO       61      84.0
6          BICICLETA       50      36.0
7                BUS       37      42.0
8       TRACTOCAMION       19       8.0
9             BUSETA       17      14.0
10          MICROBUS       17      35.0
11          VOLQUETA        7       2.0
12         MOTOCARRO        3       2.0
13          TRICICLO        1       1.0

In [ ]:
clasevehiculos= clasevehiculos.fillna(0.0) #se rellenan los nulos con ceros
clasevehiculos['TOTAL']=clasevehiculos['PALMIRA']+clasevehiculos['ENVIGADO'] #se crea una columna nueva

In [ ]:
clasevehiculos=clasevehiculos.sort_values('TOTAL', ascending = False) #se orden los datos de manera descendente de acuerdo a la columna TOTAL
clasevehiculos

CLASE DE VEHICULO  PALMIRA  ENVIGADO   TOTAL
0        MOTOCICLETA      924     536.0  1460.0
1          AUTOMOVIL      648     668.0  1316.0
3          CAMIONETA      251     257.0   508.0
2          NO APLICA      277       0.0   277.0
5            CAMPERO       61      84.0   145.0
4             CAMION       68      76.0   144.0
6          BICICLETA       50      36.0    86.0
7                BUS       37      42.0    79.0
10          MICROBUS       17      35.0    52.0
9             BUSETA       17      14.0    31.0
8       TRACTOCAMION       19       8.0    27.0
11          VOLQUETA        7       2.0     9.0
12         MOTOCARRO        3       2.0     5.0
13          TRICICLO        1       1.0     2.0

In [ ]:
# crear gráfica
fig = px.bar(clasevehiculos, x='CLASE DE VEHICULO', y=['TOTAL'], title ='<b>Accidentes segun el tipo de Vehiculo<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'TIPO DE VEHICULO',
    yaxis_title = 'ACCIDENTES',
    template = 'simple_white',
    legend_title = '<b>Cantidad<b>',
    title_x = 0.5)

fig.show()

En relación a la clase de vehículos involucrados en accidentes, se puede decir que teniendo en cuenta el grafico anterior que las motocicletas, los automóviles son los tipos que predominan en comparación a los demás, ya que ambas oscilan entre los 1300 y los 1500, ya luego le sigue camioneta con 508. 
Los demás tipos oscilan entre 0 y 145, lo cual nos dice que estos tipos de vehículos tienen poca circulación en las vías de estas ciudades.
Un dato relevante es que la categoría NO APLICA tiene una buena cantidad de datos 277, donde podemos decir que a la hora del accidente se les pasa esta información o simplemente no es relevante para el municipio correspondiente, también puede ser que se equivoquen a la hora de digitar.
Concluimos que los carros con mayor circulación en palmira y envigado son los automóviles y las motocicletas.


# PREGUNTA 7: ¿CÚAL ES LA CANTIDAD DE ACCIDENTES SEGÚN EL DISEÑO DE LA VÍA DONDE FUE EL ACCIDENTE?

In [ ]:
df9=df9.rename(columns ={'DISEÃ\x91O':'DISEÑO'})  #Se renombra el titulo de la columna 

In [ ]:
df9['DISEÑO']=df9['DISEÑO'].replace({'Pontón':'Ponton'})  #se reemplaza el nombre

In [ ]:
#se crea un dataframe con diseño y lo cuento con expediente, ordeno de manera descendente
tramo_via = df9.groupby(['DISEÑO'])[['EXPEDIENTE']].count().sort_values('EXPEDIENTE', ascending = False).rename(columns={'EXPEDIENTE':'ACCIDENTES'})
tramo_via['porcentaje'] = tramo_via.apply(lambda x: x.cumsum()/tramo_via['ACCIDENTES'].sum()) #Agregamos columna con el porcentaje acumulado segun la frecuencia de accidentes
tramo_via

ACCIDENTES  porcentaje
DISEÑO                               
Tramo de via        11835    0.600243
Interseccion         3638    0.784754
Lote o Predio        3445    0.959477
Glorieta              435    0.981539
Ciclo Ruta             93    0.986256
Paso Elevado           87    0.990668
Paso Inferior          77    0.994573
Puente                 73    0.998276
Tunel                  18    0.999189
Paso a Nivel           11    0.999746
Via peatonal            3    0.999899
Ponton                  2    1.000000

In [ ]:
# importar paquete
import plotly.graph_objects as go 

# realizo pareto
fig = go.Figure([go.Bar(x=tramo_via.index, y=tramo_via['ACCIDENTES'], yaxis='y1', name='sessions id'),
                 go.Scatter(x=tramo_via.index, y=tramo_via['porcentaje'], yaxis='y2', name='accidentes de transito', hovertemplate='%{y:.1%}', marker={'color': '#000000'})])
#agregar detalles
fig.update_layout(template='plotly_white', showlegend=False, hovermode='x', bargap=.3,
                  title={'text': '<b>Pareto accidentes de transito por diseño de la Via<b>', 'x': .5}, 
                  yaxis={'title': 'Accidentes'},
                  yaxis2={'rangemode': "tozero", 'overlaying': 'y', 'position': 1, 'side': 'right', 'title': 'Porcentaje', 'tickvals': np.arange(0, 1.1, .2), 'tickmode': 'array', 'ticktext': [str(i) + '%' for i in range(0, 101, 20)]})


fig.show()

Al momento del accidente, observamos que el tramo de vía (Zona de la vía destinada a la circulación de vehículos) o la calle como la conocemos comúnmente es donde suceden la mayoría de los accidentes, esto en comparación a los demás diseños de las vías. 
Ignorando tramo de vía, intersección y lote o predio son los que tienen la demás información significativa de estos accidentes, ambas con una diferencia casi que insignificante. 
Las glorietas se pueden tomar como significantes debido a la alta concurrencia de estas, esto ignorando las tres categorías que predominan.
Las demás categorías casi que no tienen relevancia en el estudio a la hora de realizar un enfoque ya que contienen muy pocos datos en relación a la cantidad de datos total.


# PREGUNTA 8: SEGÚN EL TIPO DE SERVICIO.¿CÚAL ES LA FRECUENCIA DE ACCIDENTALIDAD EN LA CIUDAD DE ENVIGADO?

In [ ]:
#Se crea un dateframe con tipo de servicio y lo cuento con radicado, ordeno de manera descendente
tipo_servicio=df8.groupby(['TIPO DE SERVICIO'])[['RADICADO']].count().sort_values('RADICADO', ascending = False).reset_index()
tipo_servicio['TIPO DE SERVICIO']= tipo_servicio['TIPO DE SERVICIO'].str.upper() # se transforma la columna en mayuscula

In [ ]:
# hacer la gráfica
fig = px.pie(tipo_servicio , values = 'RADICADO', names = 'TIPO DE SERVICIO', title = '<b>% ACCIDENTES SEGUN EL TIPO DE SERVICIO<b>',
             hole = .5)

cant_ed = tipo_servicio['RADICADO'].sum()


# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    legend_title = 'Tipo de Servicio',
    title_x = 0.5,
    annotations = [dict(text = str(cant_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])

fig.show()

Según el tipo de servicio, se dice que los particulares son los que predominan estos accidentes con más de 2/3 de la información, teniendo un 77.6%, esto se puede dar porque a más población más vehículos, luego le sigue servicio público que tiene concordancia con lo lógico, que es donde más vehículos además del particular se encuentran circulando las calles, este con un 17.9%.
Las demás categorías casi que no tienen mucha significancia, ya que oficial, diplomático y sin registro, están oscilando entre el 0.9% y el 2.4%, lo cual es significativamente bajo.
Los resultados resultan bastante lógicos, de acuerdo con lo que se maneja en el día a día.


# PREGUNTA 9: ¿QUÉ DIA DE LA SEMANA SE ACCIDENTAN MÁS LAS PERSONAS EN LAS TRES CIUDADES?

In [ ]:
#se agrega una columna llamada dia semana a la bodega usando la función weekday para extraer el numero del dia de la semana de la columna fecha
bodega['DIA SEMANA']=bodega['FECHA'].dt.strftime('%a')

In [ ]:
bodega

FECHA GRAVEDAD CLASE_ACCIDENTE  PALMIRA  ENVIGADO  MEDELLIN  \
0    2020-01-01    DAÑOS          choque      2.0       3.0      18.0   
1    2020-01-01  HERIDOS          choque      3.0       2.0      26.0   
2    2020-01-02    DAÑOS          choque      4.0       5.0      62.0   
3    2020-01-03    DAÑOS          choque      8.0       2.0      27.0   
4    2020-01-03  HERIDOS          choque      7.0       2.0      19.0   
...         ...      ...             ...      ...       ...       ...   
1867 2020-12-07  HERIDOS     volcamiento      0.0       0.0       2.0   
1868 2020-12-08  HERIDOS       atropello      0.0       0.0       8.0   
1869 2020-12-08  HERIDOS  caida ocupante      0.0       0.0       9.0   
1870 2020-12-08  HERIDOS            otro      0.0       0.0       7.0   
1871 2020-12-08   MUERTO       atropello      0.0       0.0       2.0   

     DIA SEMANA  
0           Wed  
1           Wed  
2           Thu  
3           Fri  
4           Fri  
...         ...  
1867        Mon  
1868        Tue  
1869        Tue  
1870        Tue  
1871        Tue  

[1872 rows x 7 columns]

In [ ]:
#se crea un dataframe con dia semana y la cuento con gravedad, lo organizo de forma descendente
bodeguita=bodega.groupby(['DIA SEMANA'])[['ENVIGADO','PALMIRA','MEDELLIN']].sum().reset_index()

In [ ]:
bodeguita['TOTAL']=(bodeguita['ENVIGADO']+bodeguita['PALMIRA']+bodeguita['MEDELLIN'])

In [ ]:
bodeguita = bodeguita.sort_values('TOTAL',ascending = False)
bodeguita

DIA SEMANA  ENVIGADO  PALMIRA  MEDELLIN   TOTAL
4        Thu     272.0    328.0    3201.0  3801.0
0        Fri     229.0    386.0    3166.0  3781.0
5        Tue     241.0    354.0    2949.0  3544.0
1        Mon     249.0    372.0    2795.0  3416.0
6        Wed     292.0    316.0    2769.0  3377.0
2        Sat     306.0    332.0    2639.0  3277.0
3        Sun     172.0    292.0    2198.0  2662.0

In [ ]:
# crear gráfica
fig = px.bar(bodeguita, x='DIA SEMANA', y=['TOTAL'], title ='<b>Cantidad de accidentes segun el Dia de la Semana<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Dia de la semana',
    yaxis_title = 'Accidentes',
    legend_title = 'Suma 3 ciudades',
    template = 'simple_white',

    title_x = 0.5)

fig.show()

Los días jueves y viernes son los dos días de la semana en los que más accidentes se registran con 3801 y 3781 accidentes respectivamente, al sumar la cantidad por día de cada ciudad. Esto puede ser debido a que llegando casi el fin de semana las personas suelen salir más, hay más tráfico y por ende pueden presentarse más accidentes, sin embargo la cantidad de accidentes en días como martes y lunes no presentan mucha diferencia respecto a los dos primeros días.


# PREGUNTA 10: ¿CÚALES SON LAS COMUNAS DE MEDELLÍN EN DONDE SE PRESENTA MAYOR ACCIDENTALIDAD?

In [ ]:
df11= df3.copy()  #se crea una copia del dataframe df3(base original de medellin)

In [ ]:
#Se renombraron algunos comunas de la columna comuna, para que queden siendo unicas
df11['COMUNA']=df11['COMUNA'].replace({'Corregimiento de San Cristóbal':'Corregimiento de San Cristobal','Corregimiento de San Sebastián de Palmitas':'Corregimiento de San Sebastian de Palmitas','La América':'La America','Belén':'Belen','0':'Sin Inf'})

In [ ]:
df11.head(2)

AÑO    CBML CLASE_ACCIDENTE      DIRECCION       DIRECCION ENCASILLADA  \
0  2020     5.0          Choque  CR 64 C CL 78  CR  064 C   078  000 00000   
1  2020  1307.0       Atropello  CL 49 CR 99 E  CL  049   099 E  000 00000   

        DISEÑO  EXPEDIENTE  FECHA_ACCIDENTE GRAVEDAD_ACCIDENTE  MES  \
0  Tramo de via  A001080297  8/01/2020 21:10        Solo daños    1   
1  Tramo de via  A001080125  8/01/2020 21:45        Con heridos    1   

   NRO_RADICADO NUMCOMUNA                  BARRIO      COMUNA  \
0       1703041         5  Terminal de Transporte    Castilla   
1       1702988        13   Juan XXIII La Quiebra  San Javier   

                          LOCATION          X           Y  
0  [-75.5729956763, 6.27615598793]  834495.75  1186012.91  
1   [-75.613806632, 6.26637688097]  829975.02  1184944.15

In [ ]:
df11= df11.drop(df11[df11['COMUNA']=='Sin Inf'].index)##Se elimina la fila sin Inf ya que sale como dato atipico en el mapa.

La Candelaria                                 3688
Laureles Estadio                              1842
Castilla                                      1692
El Poblado                                    1415
Guayabal                                      1278
Belen                                         1233
Robledo                                       1130
Aranjuez                                      1083
Buenos Aires                                   863
Manrique                                       727
Villa Hermosa                                  675
La America                                     572
Doce de Octubre                                496
Popular                                        399
Corregimiento de San Cristobal                 350
San Javier                                     337
Santa Cruz                                     308
Corregimiento de San Antonio de Prado          203
Corregimiento de Santa Elena                    89
Corregimiento de Altavista     

In [ ]:
#Se toma la columna location, se extraen los datos de longitud y latitud y se crean columnas nuevos con estos datos
df11['LONGITUD'] = df11['LOCATION'].apply (lambda x: x.strip('][').split(',')[0]).astype(float) 
df11['LATITUD'] = df11['LOCATION'].apply (lambda x: x.strip('][').split(',')[1]).astype(float)

In [ ]:
token_map = "pk.eyJ1IjoibGF1cmF5YXJjZTE5IiwiYSI6ImNsN2k5eTJuNDBtYjQzb252dnpid3hkOGsifQ.4DM4nHHs4U5PEFlsADNrPQ"
px.set_mapbox_access_token(token_map)

In [ ]:
# generar gráfica
df12 = df11.dropna(subset =['COMUNA']) # quitar valores nulos de COMUNA
px.scatter_mapbox(df12, lat = 'LATITUD', lon = 'LONGITUD', color ='COMUNA', mapbox_style  = 'streets',
                  color_continuous_scale = px.colors.cyclical.IceFire, size_max = 30, zoom = 10)

La comuna de Medellín en donde se concentra la mayor cantidad de puntos es decir accidente registrados y se puede observar que no hay espacios en donde no hayan puntos, es en la comuna número 10 que es la candelaria con un total de 3688 accidentes según el dataframe, esto se puede deber a que en la comuna candelaria se encuentras zonas altamente transitadas todos los días de la semana, tales como Bostón, Alpujarra, Bombona N1, entre otros.

# INFORMACION ADICIONAL Medellín

In [ ]:
##Base accidentalidad en el municipio de Medellin para el analisis adicional
dfm = pd.read_csv('http://medata.gov.co/sites/default/files/medata_harvest_files/mede_victimas_inci.csv',sep=';',nrows= 200000, usecols = [1,2,5,14])
#se tomaron solo 200.000 filas porque si se toman todas las filas las ultimas las carga como NaN
#se tomaron solo la columna 1,2 y 5  porque son las que tienen las categorías relevantes para complementar nuestro análisis
#se toma la columna de radicado para poder realizar las gráficas
dfm

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (14) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


Fecha_incidente Hora_incidente     Sexo Radicado
0            1/01/2014       00:15:00        M  1423940
1            1/01/2014       00:30:00        M  1423921
2            1/01/2014       00:30:00        F  1423921
3            1/01/2014       00:37:00        M  1423849
4            1/01/2014       00:37:00        M  1423849
...                ...            ...      ...      ...
199995      22/05/2020       17:30:00        M  1714854
199996      22/05/2020       17:30:00        M  1714854
199997      22/05/2020       18:30:00        F  1714852
199998      22/05/2020       19:30:00  Sin Inf  1714840
199999      22/05/2020       19:40:00        M  1714853

[200000 rows x 4 columns]

**Se tomará la hora de los accidentes, pero solo se tendrá en cuenta para la visualización las primeras 10 horas que más accidentes presenta**

NOTA: SE UTILIZARÁN LOS DATOS QUE VAN DESDE EL 01/01/2014 HASTA EL 22/05/2020.






In [ ]:
hora_acc = dfm.groupby(['Hora_incidente'])[['Radicado']].count().reset_index().rename(columns={'Radicado':'Cantidad'})
hora_acc = hora_acc.head(10) #visualizar mejor

In [ ]:
# definir gráfica
fig = px.bar(hora_acc, x='Hora_incidente', y=['Cantidad'], barmode= 'group', title ='<b>Accidentes según la hora<b>')
px.bar()

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Hora',
    yaxis_title = 'Cantidad de Accidentes',
    template = 'simple_white',
    title_x = 0.5,
    legend_title = '<b>variable<b>')


fig.show()

**Teniendo en cuenta el rango de hora que se tomó (top 10) por la cantidad de datos arrojo que la accidentalidad en Medellín se presenta entre las horas 00 y las 9am, donde las 00 horas en donde más accidentalidad se presenta con 422, donde una posible causa de esto es que a esta hora mucha gente no respeta los semáforos o se confía de la vía tan despejada y andan a una alta velocidad, a su vez la hora 5 am es la que le sigue con una accidentalidad de 262, se puede dar porque a esta hora mucha gente se dirige a su lugar de trabajo y pueden incurrir en excesos de velocidad.**

**También se tomará el sexo para complementar el análisis.**

In [ ]:
dfm['Sexo']= dfm['Sexo'].str.upper() # se pone todo en mayúscula para que todo quede en el mismo formato

In [ ]:
sex_acc = dfm.groupby(['Sexo'])[['Radicado']].count().reset_index().rename(columns={'Radicado':'Cantidad'})

#Se realiza el grafico de dataframe total
fig = px.pie(sex_acc , values = 'Cantidad', names = 'Sexo', title = '<b>% Accidentes según el sexo<b>')
# agregar detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    legend_title = 'Accidentalidad por Sexo',
    title_x = 0.5)

fig.show()


**El sexo que más tuvo accidentes de tránsito fue el masculino con un 68.7% superando por un poco más del doble al sexo contrario, muy pocos datos no cuentan con esta categoría ya que solo es un 0.62% de estos.**
